bundles/org.openhab.core/src/main/java/org/openhab/core/util/SameThreadExecutorService.java

```diff
@@ -0,0 +1,235 @@
+        Objects.requireNonNull(command);
+        if (delay == 0L) {
+            @NonNullByDefault({})
+            FutureTask<Void> future = new FutureTask<>(command, null);
+            future.run();
+            return new ImpostorScheduledFuture<Void>(future);
+        }
+        throw new UnsupportedOperationException("Delayed schedule not supported by SameThreadExecutorService");
+    }
+
+    /**
+     * <b>Not supported</b> if {@code delay} is non-zero.
+     * <p>
+     * {@inheritDoc}
+     *
+     * @throws NullPointerException if {@code callable} is {@code null}.
+     * @throws UnsupportedOperationException if {@code delay} is non-zero.
+     */
+    @Override
+    public <V> ScheduledFuture<V> schedule(@Nullable Callable<V> callable, long delay, @Nullable TimeUnit unit) {
+        Objects.requireNonNull(callable);
+        if (delay == 0L) {
+            FutureTask<V> future = new FutureTask<V>(callable);
+            future.run();
+            return new ImpostorScheduledFuture<>(future);
+        }
+        throw new UnsupportedOperationException("Delayed schedule not supported by SameThreadExecutorService");
+    }
+
+    /**
+     * <b>Will execute immediately and run only once.</b>
+     * <p>
+     * {@inheritDoc}
+     */
+    @Override
+    public ScheduledFuture<?> scheduleAtFixedRate(@Nullable Runnable task, long initialDelay, long period,
+            @Nullable TimeUnit unit) {
+        Objects.requireNonNull(task);
+        @NonNullByDefault({})
+        FutureTask<Void> future = new FutureTask<>(task, null);
+        future.run();
+        return new ImpostorScheduledFuture<Void>(future);
+    }
+
+    /**
+     * <b>Will execute immediately and run only once.</b>
+     * <p>
+     * {@inheritDoc}
+     */
+    @Override
+    public ScheduledFuture<?> scheduleWithFixedDelay(@Nullable Runnable task, long initialDelay, long delay,
+            @Nullable TimeUnit unit) {
+        Objects.requireNonNull(task);
+        @NonNullByDefault({})
+        FutureTask<Void> future = new FutureTask<>(task, null);
+        future.run();
+        return new ImpostorScheduledFuture<Void>(future);
+    }
+
+    /**
+     * A {@link ScheduledFuture} that wraps a {@link Future} and always reports zero delay.
+     *
+     * @param <T> the return type.
+     *
+     * @author Ravi Nadahar - Initial contribution
+     */
+    protected static class ImpostorScheduledFuture<T> implements ScheduledFuture<T> {
+
+        /** The wrapped {@link Future} */
+        protected final Future<T> delegate;
+
+        /**
+         * Creates a new instance that wraps the specified {@link Future}.
+         *
+         * @param future the {@link Future} to wrap.
+         */
+        public ImpostorScheduledFuture(Future<T> future) {
+            this.delegate = future;
+        }
+
+        @Override
+        public long getDelay(@Nullable TimeUnit unit) {
+            return 0L;
+        }
+
+        @Override
```

Comment:
The PMD warning (i.e. the error) should be suppressed as the impl. is intentional.

I prefer the local way over adding to the suppression list.

 ```suggestion
        @Override
        @SuppressWarnings("PMD.CompareObjectsWithEquals")
```